src/franka_plan_runner.cc

```diff
@@ -155,48 +155,45 @@ franka::RobotMode FrankaPlanRunner::GetRobotMode(franka::Robot& robot) {
 }
 
 int FrankaPlanRunner::RunFranka() {
-  //$ attempt connection to robot and read current mode
-  //$ return if connection fails, or robot is in a mode that cannot receive
-  // commands
-  try {
-    franka::Robot robot(ip_addr_);
-
-    size_t count = 0;
-    auto current_mode = GetRobotMode(robot);
-
-    if (current_mode == franka::RobotMode::kReflex) {
-      dexai::log()->warn(
-          "RunFranka: Robot in mode: {} at startup, trying to do "
-          "automaticErrorRecovery ...",
-          utils::RobotModeToString(current_mode));
-      try {
-        robot.automaticErrorRecovery();
-        dexai::log()->info(
-            "RunFranka: automaticErrorRecovery() succeeded, "
-            "robot now in mode: {}.",
+  bool connection_established {false};
+  // attempt connection to robot and read current mode
+  // and if it fails, keep trying instead of exiting the program
+  while (!connection_established) {
```

Comment:
should we have a sleep in here just in case? don't want another CPU hungry driver

Reply:
Fair point. I'll add it in